src/sql/SqlResult.ts

```diff
@@ -319,7 +317,9 @@ export class SqlResultImpl implements SqlResult {
 
         // Do not start a fetch if close is called
         if (this.closeDeferred?.promise) {
-            return Promise.reject('Cannot fetch, close() is called');
+            return Promise.reject(
```

Comment:
I think it would be much cleaner if we do this in the close method. (set the fetch promise and reject it with the error you created there). This way, we can also eliminate this branch. (`if (this.closeDeferred?.promise) {`)

Reply:
Does your suggestion cover when first close(), then fetch() called? This if check is for this case.

Reply:
yes, since in this case fetch promise will be set to something rejected (you check if it is null or not, if it is null you set it to deferredPromise(), and reject it with the error you created)